src/Wallabag/CoreBundle/Controller/ShareController.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Wallabag\CoreBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Security\Core\Exception\InvalidArgumentException;
+use Wallabag\CoreBundle\Entity\Entry;
+use Wallabag\CoreBundle\Entity\Notification;
+use Wallabag\CoreBundle\Entity\Share;
+use Wallabag\CoreBundle\Event\Activity\Actions\Entry\EntrySavedEvent;
+use Wallabag\CoreBundle\Event\Activity\Actions\Share\ShareAcceptedEvent;
+use Wallabag\CoreBundle\Event\Activity\Actions\Share\ShareCancelledEvent;
+use Wallabag\CoreBundle\Event\Activity\Actions\Share\ShareCreatedEvent;
+use Wallabag\CoreBundle\Event\Activity\Actions\Share\ShareDeniedEvent;
+use Wallabag\CoreBundle\Notifications\NoAction;
+use Wallabag\CoreBundle\Notifications\YesAction;
+use Wallabag\UserBundle\Entity\User;
+
+class ShareController extends Controller
+{
+    /**
+     * @Route("/share-user/{entry}/{destination}", name="share-entry-user", requirements={"entry" = "\d+", "destination" = "\d+"})
```

Comment:
Because we can imagine sharing something to a group, I'd rather prefer this kind of URL: `/share/user/{entry}/{destination}`.